Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIr.kt

```diff
@@ -0,0 +1,460 @@
+ */
+class SymbolicStubType(
+        val name: String,
+        val typeArguments: List<TypeArgumentStub> = emptyList(),
+        override val nullable: Boolean = false
+) : StubType()
+
+/**
+ * Represents a source of StubIr element.
+ */
+sealed class StubOrigin {
+    /**
+     * Special case when element of IR was generated.
+     */
+    object None : StubOrigin()
+
+    class ObjCMethod(
+            val method: org.jetbrains.kotlin.native.interop.indexer.ObjCMethod,
+            val container: ObjCContainer
+    ) : StubOrigin()
+
+    class ObjCClass(
+            val clazz: org.jetbrains.kotlin.native.interop.indexer.ObjCClass
+    ) : StubOrigin()
+
+    class ObjCProtocol(
+            val protocol: org.jetbrains.kotlin.native.interop.indexer.ObjCProtocol
+    ) : StubOrigin()
+
+    class Enum(val enum: EnumDef) : StubOrigin()
+
+    class Function(val function: FunctionDecl) : StubOrigin()
+
+    class FunctionParameter(val parameter: Parameter) : StubOrigin()
+
+    class Struct(val struct: StructDecl) : StubOrigin()
+}
+
+interface StubElementWithOrigin : StubIrElement {
+    val origin: StubOrigin
+}
+
+interface AnnotationHolder {
+    val annotations: List<AnnotationStub>
+}
+
+sealed class AnnotationStub {
+    sealed class ObjC : AnnotationStub() {
+        object ConsumesReceiver : ObjC()
+        object ReturnsRetained : ObjC()
+        class Method(val selector: String, val encoding: String, val isStret: Boolean = false) : ObjC()
+        class Factory(val selector: String, val encoding: String, val isStret: Boolean = false) : ObjC()
+        object Consumed : ObjC()
+        class Constructor(val selector: String, val designated: Boolean) : ObjC()
+        class ExternalClass(val protocolGetter: String = "", val binaryName: String = "") : ObjC()
+    }
+
+    sealed class CCall : AnnotationStub() {
+        object CString : CCall()
+        object WCString : CCall()
+        class Symbol(val symbolName: String) : CCall()
+    }
+
+    class CStruct(val struct: String) : AnnotationStub()
+    class CNaturalStruct(val struct: String) : AnnotationStub()
+
+    class CLength(val length: Long) : AnnotationStub()
+
+    class Deprecated(val message: String, val replaceWith: String) : AnnotationStub()
+}
+
+/**
+ * Compile-time known values.
+ */
+sealed class ConstantStub : ValueStub()
+class StringConstantStub(val value: String) : ConstantStub()
+data class IntegralConstantStub(val value: Long, val size: Int, val isSigned: Boolean) : ConstantStub()
+data class DoubleConstantStub(val value: Double, val size: Int) : ConstantStub()
+
+
+class PropertyStub(
+        val name: String,
+        val type: StubType,
+        val kind: Kind,
+        val modality: MemberStubModality = MemberStubModality.NONE,
+        val receiverType: StubType? = null,
+        override val annotations: List<AnnotationStub> = emptyList()
+) : StubIrElement, AnnotationHolder {
+    sealed class Kind {
```

Comment:
Why not have `val getter: PropertyAccessor.Getter` and `val setter: PropertyAccessor.Setter?` instead?

Reply:
In this case it is harder to distinguish `const val x: Int = 5` and `val x: Int get() = 5`. If we add field `val isConst: Boolean` to `PropertyAccessor` it would be possible to get something like `const var ...`